### PR TITLE
Limits

### PR DESCRIPTION
**Pull request by GETandSELECT**

### CODE REVIEW

File: mariadb.html.md.erb [hunk 1]
```diff
@@ -57,13 +57,36 @@ Cloud Foundry: [Spring Music Example](https://github.com/cloudfoundry-samples/sp
...
+> Dear Schema Designer:
+
+> - InnoDB only, always have PK.
+
+> Dear Developer:
+
+> - Check for errors, even after `COMMIT`.
+> - Moderate sized transactions.
+> - Don't make assumptions about `AUTO_INCREMENT` values other than uniqueness. Not monotonicity, not gapless. (This applies for non-Galera, but probably happens more often here.)
+- Handling of "critical reads" is quite different (arguably better).
```
Comment (MatthiasWinzeler) [inline mariadb.html.md.erb]: missing `>`

File: mariadb.html.md.erb [hunk 2]
```diff
@@ -57,13 +57,36 @@ Cloud Foundry: [Spring Music Example](https://github.com/cloudfoundry-samples/sp
...
+
+> Dear DBA:
+
+> - `ALTER`s are handled differently.
```
Comment (MatthiasWinzeler) [inline mariadb.html.md.erb]: how differently? needs to be clearer for the DBA/dev

File: mariadb.html.md.erb [hunk 3]
```diff
@@ -57,13 +57,36 @@ Cloud Foundry: [Spring Music Example](https://github.com/cloudfoundry-samples/sp
...
+> - `TRIGGER`s and `EVENT`s may need checking.
```
Comment (MatthiasWinzeler) [inline mariadb.html.md.erb]: what checking?

File: mariadb.html.md.erb [hunk 4]
```diff
@@ -57,13 +57,36 @@ Cloud Foundry: [Spring Music Example](https://github.com/cloudfoundry-samples/sp
...
+> - Several `VARIABLE`s need to be set differently.
```
Comment (MatthiasWinzeler) [inline mariadb.html.md.erb]: how differently? is this relevant for our platform offering?

File: mariadb.html.md.erb [hunk 5]
```diff
@@ -57,13 +57,36 @@ Cloud Foundry: [Spring Music Example](https://github.com/cloudfoundry-samples/sp
...
+
+Please add a **primary key** to every table you create. This is because of *Row Based Replication*.
+
+Please contact Swisscom Application Cloud support when you need to issue **long running DDL changes** (for example `ALTER TABLE`). Long running DDL causes an outage for all customers on the cluster.
```
Comment (MatthiasWinzeler) [inline mariadb.html.md.erb]: should they contact our support for each `ALTER TABLE` they run? Probably not, but it sounds like this. What is long running? Do we really wanna be contacted for this? What do we do about it, anyway?

File: mariadb.html.md.erb [hunk 6]
```diff
@@ -57,13 +57,36 @@ Cloud Foundry: [Spring Music Example](https://github.com/cloudfoundry-samples/sp
...
+
+Our Galera aaS is a **shared cluster**. Each customer has their own database. Put simply, the service broker executes the following statements:
 
 ```sql
 CREATE DATABSE CF_DATABASE_NAME
-GRANT ALL PRIVILEGES ON `CF_DATABASE_NAME`.* TO 'USER'@'SUBNET'
+GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, REFERENCES, INDEX, ALTER, CREATE TEMPORARY TABLES, EXECUTE, CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, EVENT, TRIGGER ON `CF_DATABASE_NAME`.* TO 'USER'@'SUBNET'
```
Comment (MatthiasWinzeler) [inline mariadb.html.md.erb]: We don't really do this, but do a `GRANT ALL` followed by a `REVOKE LOCK TABLES` instead.

File: mariadb.html.md.erb (hunk 3 above)
Comment (GETandSELECT) [reply mariadb.html.md.erb]: I don't know myself. I will investigate and update text

**Pull request merged**